src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/InterceptedProjectProperties.cs

```diff
@@ -98,5 +110,34 @@ public void SetRuleContext(Rule rule)
             var ruleAwareProperties = DelegatedProperties as IRuleAwareProjectProperties;
             ruleAwareProperties?.SetRuleContext(rule);
         }
+
+        private IInterceptingPropertyValueProvider? GetFilteredProvider(string propertyName, (List<Lazy<IInterceptingPropertyValueProvider, IInterceptingPropertyValueProviderMetadata>> Providers, bool Filtered) valueTuple)
+        {
+            if (valueTuple.Filtered)
+            {
+                return ReturnProviderAfterFiltering();
+            }
+
+            valueTuple.Providers.RemoveAll(lazyProvider =>
+            {
+                string? appliesToExpression = lazyProvider.Value.GetType().GetCustomAttributes(true).OfType<AppliesToAttribute>().FirstOrDefault()?.AppliesTo;
```

Comment:
```suggestion
                string? appliesToExpression = lazyProvider.Value.GetType().GetCustomAttributes(typeof(AppliesToAttribute), inherit: true).OfType<AppliesToAttribute>().FirstOrDefault()?.AppliesTo;
```

- named argument
- specify type of interest to reduce work under the covers

This line's getting quite long so you may like to wrap it.